File: Behavioral/Command/CSharp/RemoteControl.cs
// RemoteControl.cs

using System;

public class RemoteControl {
  private ICommand? _command;

  public void SetCommand(ICommand command) {
    _command = command;
  }

  public void PressButton() {
    if (_command != null) {
      _command.Execute();
    } else {
      Console.WriteLine("No command assigned to button.");
    }
  }
}

File: Behavioral/Command/JavaScript/remoteControl.js
// remoteControl.js

class RemoteControl {
  constructor() {
    this.command = null;
  }

  setCommand(command) {
    this.command = command;
  }

  pressButton() {
    if (this.command) {
      this.command.execute();
    } else {
      console.log("No command assigned to button.");
    }
  }
}

module.exports = RemoteControl;
